xbmc/video/VideoInfoScanner.cpp

```diff
@@ -298,6 +298,11 @@ namespace VIDEO
       { // need to fetch the folder
         CDirectory::GetDirectory(strDirectory, items, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
                                  DIR_FLAG_DEFAULTS);
+        // do not consider inner folders with .nomedia
+        items.erase(std::remove_if(items.begin(), items.end(),
+                                   [&](const CFileItemPtr& item)
```

Comment:
enough to capture `[this]` for the HasNoMedia() call.

Reply:
I assume that passes this by reference right?

Reply:
no by val, but it‘s only a pointer and you can‘t change the value of `this` anyway